Cluster nodes command does not seem to be 100% redis compatible

**Describe the bug**
We are running flink applications using the lettuce redis library.
It seems to struggle with the output of dragonflies response to the `cluster nodes` command.
Somehow it seems there is a difference to what it expects.

Output from dragonfly:
```
$ redis-cli cluster nodes
807df88cb947566738e9fa4689231944385a353b 127.0.0.1:6379@6379 myself,master - 0 0 0 connected 0-16383
c23888188f5d350b552aa8d8aa7ad40a05765b26 10.37.159.206:6379@6379 slave 807df88cb947566738e9fa4689231944385a353b 0 0 0 connected
c23888188f5d350b552aa8d8aa7ad40a05765b26 10.36.8.23:6379@6379 slave 807df88cb947566738e9fa4689231944385a353b 0 0 0 connected
```

Output from redis:
```
$ redis-cli cluster nodes
f7b5cf36a5ffc927a2de8b507b880e8cd60f45d3 10.36.54.105:6379@16379,redis-app-leader-2 master - 1710182388055 1710182385544 3 connected
09a46f18f4a8077ced90f4a394fc4f73ecc206e3 10.38.135.168:6379@16379,redis-app-leader-1 master - 0 1710413134000 2 connected 5461-10922
3991f305a18628b81d61e4040206ade513145f39 10.37.234.11:6379@16379,redis-app-leader-0 master - 0 1710413134727 1 connected 0-5460
2ddedc9d0cf086539b81ba1b22c88a7ec35a4f88 10.38.174.196:6379@16379,redis-app-follower-0 myself,slave 3991f305a18628b81d61e4040206ade513145f39 0 1710413133000 1 connected
273c6030cf5044e27675919b5317e3eeec880e91 10.36.123.79:6379@16379,redis-app-follower-2 master - 0 1710413134000 4 connected 10923-16383
147b1aa03a7f4552098f3139e8fb0924e6a20689 10.37.195.111:6379@16379,redis-app-follower-1 slave 09a46f18f4a8077ced90f4a394fc4f73ecc206e3 0 1710413135028 2 connected
```

The exception from lettuce:
```
io.lettuce.core.RedisConnectionException: Unable to establish a connection to Redis Cluster
	at io.lettuce.core.cluster.RedisClusterClient.lambda$assertInitialPartitions$26(RedisClusterClient.java:922)
	at io.lettuce.core.cluster.RedisClusterClient.get(RedisClusterClient.java:941)
	at io.lettuce.core.cluster.RedisClusterClient.assertInitialPartitions(RedisClusterClient.java:921)
	at io.lettuce.core.cluster.RedisClusterClient.connect(RedisClusterClient.java:398)
	at io.lettuce.core.cluster.RedisClusterClient.connect(RedisClusterClient.java:375)
	at io.justtrack.flink.annotators.repositories.StoreRedis.connect(StoreRedis.java:205)
	at io.justtrack.flink.annotators.repositories.StoreRedis.getRedis(StoreRedis.java:173)
	at io.justtrack.flink.annotators.repositories.StoreRedis.put(StoreRedis.java:108)
	... 7 more
Caused by: io.lettuce.core.RedisException: Cannot parse 29bc106e11efccc5b3db4e3397511a0539c74b10 10.37.102.233:6379@6379 myself,master - 0 0 0 connected 0-16383
d4d9fd811e189f246d8c591fb10652f993933f39 10.36.112.239:6379@6379 slave 29bc106e11efccc5b3db4e3397511a0539c74b10 0 0 0 connected
d4d9fd811e189f246d8c591fb10652f993933f39 10.38.180.168:6379@6379 slave 29bc106e11efccc5b3db4e3397511a0539c74b10 0 0 0 connected

	at io.lettuce.core.cluster.models.partitions.ClusterPartitionParser.parse(ClusterPartitionParser.java:89)
	at io.lettuce.core.cluster.topology.NodeTopologyView.<init>(NodeTopologyView.java:68)
	at io.lettuce.core.cluster.topology.NodeTopologyView.from(NodeTopologyView.java:90)
	at io.lettuce.core.cluster.topology.DefaultClusterTopologyRefresh.getNodeSpecificViews(DefaultClusterTopologyRefresh.java:229)
	at io.lettuce.core.cluster.topology.DefaultClusterTopologyRefresh.lambda$null$1(DefaultClusterTopologyRefresh.java:108)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.DefaultEventExecutor.run(DefaultEventExecutor.java:66)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
Caused by: java.lang.NumberFormatException: For input string: "16383
"
	at java.base/java.lang.NumberFormatException.forInputString(Unknown Source)
	at java.base/java.lang.Integer.parseInt(Unknown Source)
	at java.base/java.lang.Integer.parseInt(Unknown Source)
	at io.lettuce.core.cluster.models.partitions.ClusterPartitionParser.readSlots(ClusterPartitionParser.java:173)
	at io.lettuce.core.cluster.models.partitions.ClusterPartitionParser.parseNode(ClusterPartitionParser.java:133)
	at io.lettuce.core.cluster.models.partitions.ClusterPartitionParser.parse(ClusterPartitionParser.java:85)
	... 12 more
```

**Expected behavior**
redis' and dragonfly should have compatible output

**Environment (please complete the following information):**
 - OS: linux
 - Kernel: 6.1
 - Containerized: Kubernetes
 - Dragonfly Version: v1.15.1

--- a/src/server/cluster/cluster_family.cc
+++ b/src/server/cluster/cluster_family.cc
@@ -250,7 +250,8 @@ void ClusterNodesImpl(const ClusterShards& config, string_view my_id, Connection
       }
     }
 
-    absl::StrAppend(&result, "\r\n");
+    // Separate lines with only \n, not \r\n, see #2726
+    absl::StrAppend(&result, "\n");
   };
 
   for (const auto& shard : config) {
@@ -280,6 +281,7 @@ namespace {
 void ClusterInfoImpl(const ClusterShards& config, ConnectionContext* cntx) {
   std::string msg;
   auto append = [&msg](absl::AlphaNum a1, absl::AlphaNum a2) {
+    // Separate lines with \r\n, not \n, see #2726
     absl::StrAppend(&msg, a1, ":", a2, "\r\n");
   };
 

--- a/src/server/cluster/cluster_family_test.cc
+++ b/src/server/cluster/cluster_family_test.cc
@@ -188,7 +188,7 @@ TEST_F(ClusterFamilyTest, ClusterConfigNoReplicas) {
                                         "abcd1234")))));
 
   EXPECT_EQ(Run({"cluster", "nodes"}),
-            "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-16383\r\n");
+            "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-16383\n");
 }
 
 TEST_F(ClusterFamilyTest, ClusterConfigFull) {
@@ -259,8 +259,8 @@ TEST_F(ClusterFamilyTest, ClusterConfigFull) {
                                         "wxyz")))));
 
   EXPECT_EQ(Run({"cluster", "nodes"}),
-            "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-16383\r\n"
-            "wxyz 10.0.0.10:8000@8000 slave abcd1234 0 0 0 connected\r\n");
+            "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-16383\n"
+            "wxyz 10.0.0.10:8000@8000 slave abcd1234 0 0 0 connected\n");
 }
 
 TEST_F(ClusterFamilyTest, ClusterConfigFullMultipleInstances) {
@@ -383,10 +383,10 @@ TEST_F(ClusterFamilyTest, ClusterConfigFullMultipleInstances) {
                                             "qwerty")))))));
 
   EXPECT_THAT(Run({"cluster", "nodes"}),
-              "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-10000\r\n"
-              "wxyz 10.0.0.10:8000@8000 slave abcd1234 0 0 0 connected\r\n"
-              "efgh7890 10.0.0.2:7001@7001 master - 0 0 0 connected 10001-16383\r\n"
-              "qwerty 10.0.0.11:8001@8001 slave efgh7890 0 0 0 connected\r\n");
+              "abcd1234 10.0.0.1:7000@7000 master - 0 0 0 connected 0-10000\n"
+              "wxyz 10.0.0.10:8000@8000 slave abcd1234 0 0 0 connected\n"
+              "efgh7890 10.0.0.2:7001@7001 master - 0 0 0 connected 10001-16383\n"
+              "qwerty 10.0.0.11:8001@8001 slave efgh7890 0 0 0 connected\n");
 
   absl::InsecureBitGen eng;
   while (true) {
@@ -818,7 +818,7 @@ TEST_F(ClusterFamilyEmulatedTest, ClusterSlots) {
 
 TEST_F(ClusterFamilyEmulatedTest, ClusterNodes) {
   EXPECT_THAT(Run({"cluster", "nodes"}),
-              GetMyId() + " fake-host:6379@6379 myself,master - 0 0 0 connected 0-16383\r\n");
+              GetMyId() + " fake-host:6379@6379 myself,master - 0 0 0 connected 0-16383\n");
 }
 
 }  // namespace

